prover/src/witness/traces.rs

```diff
@@ -151,38 +150,45 @@ impl<T: Copy> Traces<T> {
             poseidon_sponge_ops,
         } = self;
 
-        let arithmetic_trace = timed!(
-            timing,
-            "generate arithmetic trace",
-            all_stark.arithmetic_stark.generate_trace(arithmetic_ops)
-        );
-        let cpu_rows: Vec<_> = cpu.into_iter().map(|x| x.into()).collect();
-        let cpu_trace = trace_rows_to_poly_values(cpu_rows);
-        let poseidon_trace = timed!(
-            timing,
-            "generate Poseidon trace",
-            all_stark
-                .poseidon_stark
-                .generate_trace(poseidon_inputs, min_rows, timing)
-        );
-        let poseidon_sponge_trace = timed!(
-            timing,
-            "generate Poseidon sponge trace",
-            all_stark
-                .poseidon_sponge_stark
-                .generate_trace(poseidon_sponge_ops, min_rows, timing)
-        );
-        let logic_trace = timed!(
-            timing,
-            "generate logic trace",
-            all_stark
-                .logic_stark
-                .generate_trace(logic_ops, min_rows, timing)
-        );
-        let memory_trace = timed!(
-            timing,
-            "generate memory trace",
-            all_stark.memory_stark.generate_trace(memory_ops, timing)
+        let mut memory_trace= vec![];
+        let mut arithmetic_trace= vec![];
+        let mut cpu_trace= vec![];
+        let mut poseidon_trace= vec![];
+        let mut poseidon_sponge_trace= vec![];
+        let mut logic_trace = vec![];
+
+        rayon::join(
```

Comment:
we need to estimate the time for the whole rayon::join.

Reply:
ok